models/ipfs_networks.go

```diff
@@ -44,14 +45,26 @@ func (im *IPFSNetworkManager) GetAPIURLByName(name string) (string, error) {
 	return pnet.APIURL, nil
 }
 
-// TODO: Validate swarm key and API url
+func (im *IPFSNetworkManager) UpdateNetwork(network *HostedIPFSPrivateNetwork) error {
+	check := im.DB.Save(network)
```

Comment:
One thing to keep in mind, `Save` will also update fields that haven't been updated. Say for example `BootStrapPeerIDs` was empty, it would then overwrite the database with an empty `BootStrapPeerIDs`

Reply:
Hm... I think I'll remove this for now then. It works over at ipfs-orchestrator because I pull data into a struct and use that same struct for the save, but having it as part of this manager API doesnt make the potentially destructive nature of this function very clear.